layer1/executor/task_handler.go

```diff
@@ -0,0 +1,107 @@
+package executor
+
+import (
+	"context"
+	"github.com/alicenet/alicenet/consensus/db"
+	"github.com/alicenet/alicenet/constants"
+	"github.com/alicenet/alicenet/layer1"
+	"github.com/alicenet/alicenet/layer1/executor/tasks"
+	monitorInterfaces "github.com/alicenet/alicenet/layer1/monitor/interfaces"
+	"github.com/alicenet/alicenet/layer1/transaction"
+	"github.com/alicenet/alicenet/logging"
+	"github.com/google/uuid"
+	"github.com/sirupsen/logrus"
+)
+
+var _ TaskHandler = &Handler{}
+
+type Handler struct {
+	manager          *TaskManager
+	logger           *logrus.Entry
+	closeMainContext context.CancelFunc
+	requestChannel   chan managerRequest
+}
+
+// NewTaskHandler creates a new Handler instance.
+func NewTaskHandler(database *db.Database, eth layer1.Client, contracts layer1.AllSmartContracts, adminHandler monitorInterfaces.AdminHandler, txWatcher transaction.Watcher) (TaskHandler, error) {
+	// main context that will cancel all workers and go routines
+	mainCtx, cf := context.WithCancel(context.Background())
```

Comment:
cf() not called on all exit paths
This is a memory leak!

Reply:
Given this is a constructor the cf should be stored on the object and called on close OR the cf should not be put in place.

Reply:
The cf is stored on the Handler object (L:43) and it is called on the close function (L:60). If the TaskManager or TaskExecutor creation fails the cf is called right after that (L:36). Is there any other exit paths I'm missing?